protocol/src/main/java/io/zeebe/protocol/Protocol.java

```diff
@@ -69,4 +69,10 @@ public static long encodePartitionId(final int partitionId, final long key) {
   public static int decodePartitionId(final long key) {
     return (int) (key >> KEY_BITS);
   }
+
+  public static long decodeKeyInPartition(final long key) {
+    final long partitionId = decodePartitionId(key);
+
+    return key - ((long) partitionId << KEY_BITS);
+  }
```

Comment:
I think we can simplify this via bit wise operator `key & 0x0007FFFFFFFFFFFF`

```python
>>> 2251799813685260 & 0x0007FFFFFFFFFFFF
12
```

How does it work:

```
0000 0000 0000 1000 0000 .... 0001 -> Key 1 and partition 1 = 2251799813685249
            & 
0000 0000 0000 0111  1111  .... 1111 = mask 51 1 bits
            =
0000 0000 0000 0000  0000 ... 0001 = 1
```